sources/packages/backend/apps/api/src/route-controllers/assessment/_tests_/e2e/assessment.institutions.controller.getAssessmentAwardDetails.e2e-spec.ts

```diff
@@ -0,0 +1,257 @@
+    finalAwards[`disbursementReceipt1Id`] = disbursementReceipt.id;
+
+    disbursementReceiptsValues.forEach((disbursementReceiptsValue) => {
+      finalAwards[
+        `disbursementReceipt1${disbursementReceiptsValue.grantType.toLowerCase()}`
+      ] = disbursementReceiptsValue.grantAmount;
+    });
+
+    const endpoint = `/institutions/assessment/student/${student.id}/assessment/${assessment.id}/award`;
+    const institutionUserToken = await getInstitutionToken(
+      InstitutionTokenTypes.CollegeFUser,
+    );
+
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(institutionUserToken, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK)
+      .expect({
+        applicationNumber: application.applicationNumber,
+        applicationStatus: application.applicationStatus,
+        institutionName:
+          assessment.offering.educationProgram.institution.operatingName,
+        offeringIntensity: assessment.offering.offeringIntensity,
+        offeringStudyStartDate: getDateOnlyFormat(
+          assessment.offering.studyStartDate,
+        ),
+        offeringStudyEndDate: getDateOnlyFormat(
+          assessment.offering.studyEndDate,
+        ),
+        estimatedAward: {
+          disbursement1Date: getDateOnlyFormat(
+            firstDisbursementSchedule.disbursementDate,
+          ),
+          disbursement1Status:
+            firstDisbursementSchedule.disbursementScheduleStatus,
+          disbursement1COEStatus: firstDisbursementSchedule.coeStatus,
+          disbursement1MSFAANumber: MASKED_MSFAA_NUMBER,
+          disbursement1MSFAAId: firstDisbursementSchedule.msfaaNumber.id,
+          disbursement1MSFAACancelledDate:
+            firstDisbursementSchedule.msfaaNumber.cancelledDate,
+          disbursement1MSFAADateSigned:
+            firstDisbursementSchedule.msfaaNumber.dateSigned,
+          disbursement1TuitionRemittance:
+            firstDisbursementSchedule.tuitionRemittanceRequestedAmount,
+          disbursement1Id: firstDisbursementSchedule.id,
+          disbursement1DocumentNumber: firstDisbursementSchedule.documentNumber,
+          ...awards,
+        },
+        finalAward: finalAwards,
+      });
+  });
+
+  it("Should throw not found exception when assessment is not found.", async () => {
+    // Arrange
+
+    // Student.
+    const student = await saveFakeStudent(db.dataSource);
+    const currentMSFAA = createFakeMSFAANumber(
+      { student },
+      {
+        state: MSFAAStates.Signed,
+      },
+    );
+    await db.msfaaNumber.save(currentMSFAA);
+    await saveFakeApplicationDisbursements(db.dataSource, {
+      institution: collegeF,
+      institutionLocation: collegeFLocation,
+      student,
+      msfaaNumber: currentMSFAA,
+    });
+
+    const institutionUserToken = await getInstitutionToken(
+      InstitutionTokenTypes.CollegeFUser,
+    );
+
+    const endpoint = `/institutions/assessment/student/${student.id}/assessment/9999999/award`;
```

Comment:
When I looked at the API while reviewing the test case, I found something.

If a student has submitted an application for an institution, then the institution can see the assessments of the student which does not belong to the institution also right? - Say if a student submitted app1, asm1 for an institution i1 and then submitted app2, asm2 for an institution i2.

i1 can see the assessment details of i2 via this API. Correct me if wrong.

![image](https://github.com/bcgov/SIMS/assets/54600590/4b68682e-4548-4a61-97d8-81539a6e38ab)

Reply:
I see the point, it is a nice catch and I believe that we should raise it to @JasonCTang.
That said, it still respects the authorization rule to allow one institution to have access to the student data once they had some relation with the student. Unless I am wrong, when we talked about the assessment history, for instance, we raised the point that the institution would be able to see all assessments, even from other institutions and I believe that the answer was that it would be the expectation.

Reply:
Great catch, if that is the requirement, we can add additional test cases for that scenario for institutions peeking.

Reply:
I see the point. Ya we can confirm with @JasonCTang and i will take it in PR next